### DISCUSSION TITLE
Extract Images from a PDF and compute checksums for those images

### DISCUSSION THREAD

**Discussion by MeowSaiGithub** [pdfcpu/pdfcpu#893] [Q&A] [answered]
Hello, I am trying to extract images from a pdf and compute checksums from those extracted images, if i extract the images and save as files with ExtractImagesFile(), and then read the bytes again from those images, won't this create a load on I/O, can i get the bytes directly from extracted images themself. There is ExtractImagesRaw() but I dont know how to use that properly. Can anyone has any suggestions. Thank you.

**Comment by hhrutter** [comment, marked as answer]
Please check `TestExtractImagesLowLevel` in extract_test.go:

<img width="1112" alt="Screenshot 2024-06-20 at 09 18 28" src="https://github.com/pdfcpu/pdfcpu/assets/11322155/6be43ac9-1577-4ea3-a554-06f00d14a363">

**Comment by MeowSaiGithub** [comment]
Thank you for your quick answer. I saw that file when i was researching a way. To my understanding, it write as a file again. How can i get []bytes or img.Bounds() so I can compute sha256 checksum.

**Comment by hhrutter** [comment]
`pdfcpu.ExtractImages` returns a slice of `model.Image` which in turn is an `io.Reader`.

**Reply by MeowSaiGithub** [reply]
thank you. that point me to a good direction
